### PR TITLE
Add Once-for-all to pytorchhub

### PR DESCRIPTION
**Pull request by Lyken17**
[Once-for-all](https://arxiv.org/abs/1908.09791) is a techinique that allows people to de-couple training and search thus enables quickly arhitecture specialization for diverse hardware platforms. It has received 559 citations since published and this PR aims to add the usage demo into torchub.

### CODE REVIEW

File: pytorch_vision_once_for_all.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+<!-- Only change fields below(remove this line before submitting a PR). Take inspiration e.g. from pytorch_vision_fcn_resnet101.md -->
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:6]: Let's remove this line :)

File: pytorch_vision_once_for_all.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+More models and configurations can be found in [once-for-all/model-zoo](https://github.com/mit-han-lab/once-for-all#evaluate-1)
+and obtained through following scripts
+
+```
+ofa_specialized_get = torch.hub.load('mit-han-lab/once-for-all', "ofa_specialized_get")
+model = ofa_specialized_get("flops@595M_top1@80.0_finetune@75", pretrained=True)
+```
+
+The model's prediction can evalutaed by
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:81]: can be evaluated

File: pytorch_vision_once_for_all.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+#    resnet50D_MAC@1.8B_top1@78.3
+#    resnet50D_MAC@1.2B_top1@77.1_finetune@25
+#    resnet50D_MAC@0.9B_top1@76.3_finetune@25
+#    resnet50D_MAC@0.6B_top1@75.0_finetune@25
+
+specialized_net = torch.hub.load('mit-han-lab/once-for-all', net_config, pretrained=True).eval()
+```
+
+More models and configurations can be found in [once-for-all/model-zoo](https://github.com/mit-han-lab/once-for-all#evaluate-1)
+and obtained through following scripts
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:74]: through the following

File: pytorch_vision_once_for_all.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+```
+ofa_specialized_get = torch.hub.load('mit-han-lab/once-for-all', "ofa_specialized_get")
+model = ofa_specialized_get("flops@595M_top1@80.0_finetune@75", pretrained=True)
+```
+
+The model's prediction can evalutaed by 
+```
+# Download an example image from the pytorch website
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:83]: from pytorch website

**Review by vmoens — COMMENTED** [review]
Can you address these comments? Thanks!

File: pytorch_vision_once_for_all.md (hunk 1 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md]: Done

File: pytorch_vision_once_for_all.md (hunk 2 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md]: Grammar fixed.

File: pytorch_vision_once_for_all.md (hunk 3 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md]: Grammar fixed.

**Comment by Lyken17** [comment]
Comments addressed. Please have a check :)

File: pytorch_vision_once_for_all.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: Once for All: Train One Network and Specialize it for Efficient Deployment
+summary: We propose to train a once-for-all (OFA) network that supports diverse architectural settings by decoupling training and search, thus achieving efficient inference across many devices and resource constraints, especially on edge devices.
+image: https://github.com/mit-han-lab/once-for-all/raw/master/figures/overview.png
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:8]: perhaps let's put the images inside the PR?

File: pytorch_vision_once_for_all.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+author: MIT Han Lab
+tags: [vision]
+github-link: https://github.com/mit-han-lab/once-for-all
+github-id: https://github.com/mit-han-lab
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:12]: should be `mit-han-lab/once-for-all`

File: pytorch_vision_once_for_all.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: Once for All: Train One Network and Specialize it for Efficient Deployment
+summary: We propose to train a once-for-all (OFA) network that supports diverse architectural settings by decoupling training and search, thus achieving efficient inference across many devices and resource constraints, especially on edge devices.
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:7]: Could it be much shorter? Like 10-15 words max?

File: pytorch_vision_once_for_all.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+image: https://github.com/mit-han-lab/once-for-all/raw/master/figures/overview.png
+author: MIT Han Lab
+tags: [vision]
+github-link: https://github.com/mit-han-lab/once-for-all
+github-id: https://github.com/mit-han-lab
+featured_image_1: https://github.com/mit-han-lab/once-for-all/raw/master/figures/cnn_imagenet_new.png
+featured_image_2: no-image
+accelerator: cuda-optional
+---
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:16]: This section is not displayed as a table for a reason that I can't really understand. The results is that it does not appear on the pre-built page on netlify (see tests).
When opening the .md file on github, you should see the first section as a table.

**Review by vmoens — COMMENTED** [review]
Some other comments.
Also, are you sure you want to name this file pytorch_vision?
I think this name is reserved to models that can be found on torchvision repo.

File: pytorch_vision_once_for_all.md (hunk 5 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md:8]: OK, moved.

File: pytorch_vision_once_for_all.md (hunk 6 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md:12]: fixed.

File: pytorch_vision_once_for_all.md (hunk 7 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md:7]: Sure, how about current version?

File: pytorch_vision_once_for_all.md (hunk 8 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md]: Fixed. This is because the missing quote on the title.

**Comment by Lyken17** [comment]
> Also, are you sure you want to name this file pytorch_vision?I think this name is reserved to models that can be found on torchvision repo.

 Our lab previous contributed a repo [pytorch_vision_proxylessnas](https://github.com/pytorch/hub/blob/master/pytorch_vision_proxylessnas.md). Does the naming pattern change recently?

File: pytorch_vision_once_for_all.md [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+title: "Once for All: Train One Network and Specialize it for Efficient Deployment"
+category: researchers
+summary: Once-for-all (OFA) decouples training and search, and achieves efficient inference across various edge devices and resource constraints.
+image: https://github.com/mit-han-lab/once-for-all/raw/master/figures/overview.png
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:8]: This should be a logo displayed on hub's main page
have a look here
https://deploy-preview-293--pytorch-hub-preview.netlify.app/hub/research-models

Put it in torchhub `image` folder and just put the name of the logo (have a look at how [gpunet](https://github.com/pytorch/hub/blob/master/nvidia_deeplearningexamples_gpunet.md) does it)

File: pytorch_vision_once_for_all.md [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+```
+
+
+### Model Description
+Once-for-all models are from [Once for All: Train One Network and Specialize it for Efficient Deployment](https://arxiv.org/abs/1908.09791)
+
+
+Conventional approaches either manually design or use neural architecture search (NAS) to find a specialized neural network and train it from scratch for each case, which is computationally prohibitive (causing CO2 emission as much as 5 cars' lifetime) thus unscalable. In this work, we propose to train a once-for-all (OFA) network that supports diverse architectural settings by decoupling training and search. Across diverse edge devices, OFA consistently outperforms state-of-the-art (SOTA) NAS methods (up to 4.0% ImageNet top1 accuracy improvement over MobileNetV3, or same accuracy but 1.5x faster than MobileNetV3, 2.6x faster than EfficientNet w.r.t measured latency) while reducing many orders of magnitude GPU hours and CO2 emission. In particular, OFA achieves a new SOTA 80.0% ImageNet top-1 accuracy under the mobile setting (<600M MACs).
+
+![](https://github.com/mit-han-lab/once-for-all/raw/master/figures/cnn_imagenet_new.png)
```
Comment (vmoens) [inline pytorch_vision_once_for_all.md:128]: this image is extremenly big
Also it depends on the master branch, meaning that if you update the repo and the image is not there anymore torch hub display will be messed up. I would suggest to upload all the content you need (images etc) on torch hub git repo.

File: pytorch_vision_once_for_all.md (hunk 9 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md:8]: Sure, moved.

File: pytorch_vision_once_for_all.md (hunk 10 above)
Comment (Lyken17) [reply pytorch_vision_once_for_all.md:128]: OK, moved.

File: pytorch_vision_once_for_all.md [hunk 11]
```diff
@@ -0,0 +1,142 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+title: "Once-for-All"
+category: researchers
+summary: Once-for-all (OFA) decouples training and search, and achieves efficient inference across various edge devices and resource constraints.
+image: once_for_all_overview.png
```
[the remark this answers is not in the document]
Comment (Lyken17) [reply pytorch_vision_once_for_all.md:8]: I prefer use the paper figures as logo. So people can pay more attention to the work itself, rather than the instituions.

[the remark this answers is not in the document]
Comment (vmoens) [reply pytorch_vision_once_for_all.md:8]: Yeah if your `[mit_hanlab_logo.png](https://github.com/pytorch/hub/pull/293/files#diff-f5f048307d024cfa535b8d157925192ef5b41d6c30050be6cb4249f896d5cd3e)` logo is not on any page perhaps we can remove it

**Review by vmoens — APPROVED** [review]
LGTM thanks for contributing!
